api/envoy/config/core/v3/base.proto

```diff
@@ -313,6 +313,17 @@ message HeaderMap {
   repeated HeaderValue headers = 1;
 }
 
+// A directory that is watched for changes, e.g. by inotify on Linux.
+message WatchedDirectory {
+  // Path to watch. Any move events in this directory will trigger an update.
+  string root = 1 [(validate.rules).string = {min_len: 1}];
+
+  // A subdirectory from *root* containing watched files. This will be resolved
+  // via readlink(2) of *root*/*subdirectory* on an inotify event prior to
+  // reading files.
+  string subdirectory = 2;
```

Comment:
As we discussed offline, it seems a little strange to me to have to specify the parent directory and the subdirectory separately.  It seems like it should be possible to just directly monitor the subdirectory (which ideally would work regardless of whether or not it's a symlink).

Reply:
Updated to an interface providing more control.